warehouse/query-core/src/main/java/datawave/query/jexl/visitors/EventDataQueryExpressionVisitor.java

```diff
@@ -167,78 +182,43 @@ private boolean apply(Key key, boolean update) {
                 final Set<Type> types = EventDataQueryExpressionVisitor.extractTypes(attributeFactory, keyFieldName, keyFieldValue, key);
                 // always add the NoOpType to ensure the original value gets propagated through
                 types.add(new NoOpType(keyFieldValue));
-                final Set<Matcher> normalizedPatternMatchers = new HashSet<>();
-                final Set<String> normalizedFieldValues = new HashSet<>();
-                final Set<LiteralRange> normalizedRanges = new HashSet<>();
+                
                 for (Type type : types) {
-                    // normalize all patterns
-                    for (Pattern fieldPattern : fieldPatterns.keySet()) {
-                        try {
-                            String normalizedPattern = type.normalizeRegex(fieldPattern.toString());
-                            if (normalizedPattern != null) {
-                                normalizedPatternMatchers.add(Pattern.compile(normalizedPattern).matcher(EMPTY_STRING));
-                            } else {
-                                // can't normalize so add the original matcher
-                                normalizedPatternMatchers.add(fieldPatterns.get(fieldPattern));
-                            }
-                        } catch (Exception e) {
-                            // can't normalize this pattern, add the original matcher
-                            normalizedPatternMatchers.add(fieldPatterns.get(fieldPattern));
-                        }
+                    
+                    boolean seenType = seenTypes.contains(type);
```

Comment:
The following operations only execute if `seenType` is false. Additionally, `fieldValues`, `fieldPatterns`, and `fieldRanges` are initialized in the constructor, so the check for null is unecessary. Thus, this section can be simplified to:

```java

if(!seenTypes.contains(type)) {
	
    // Check for value normalization
    if(!fieldValues.isEmpty())
        allNormalizedValues.addAll(normalizeValuesForType(type));

    // Check for pattern normalization
    if (!fieldPatterns.isEmpty())
        allNormalizedPatterns.addAll(normalizePatternsForType(type));

    // Check for range normalization
    if(!fieldRanges.isEmpty())
        allNormalizedRanges.addAll(normalizeRangesForType(type));

    // Add this type to seen types.
    seenTypes.add(type);
}
```

Reply:
Good suggestion, implemented while incorporated while resolving merge conflicts from bounded range changes.